### PR TITLE
Update client.py by @tharris72

### PR DESCRIPTION
**Pull request by jsignell**
> Adding the Returns numpy style documentation to the doc strings. This is for issue #3578.
Testing out the pull request to make sure it works. There are still more doc strings to update.

This replaces #4517 since that PR got pretty stale. I'll merge this today if there are now complaints.

### CODE REVIEW

**Comment by jsignell** [comment]
Thanks  @tharris72 this is in!

File: distributed/client.py [hunk 1]
```diff
@@ -212,21 +214,55 @@ def __init__(self, key, client=None, inform=True, state=None):
...
+        Raises
+        ------
+        dask.distributed.TimeoutError
+            If *timeout* seconds are elapsed before returning, a
+            ``dask.distributed.TimeoutError`` is raised.
+
+        Returns
+        -------
+        result : asyncio.Future
+            The Future that contains the result of the computation
```
Comment (jrbourbeau) [inline distributed/client.py:265]: I think this is be incorrect. `Future.result()` should return the actual result (or a coroutine if the `Client` is in asynchronous mode).

File: distributed/client.py [hunk 2]
```diff
@@ -354,6 +418,11 @@ def traceback(self, timeout=None, **kwargs):
         >>> traceback.format_tb(tb)  # doctest: +SKIP
         [...]
 
+        Returns
+        -------
+        Future
+            The Future that contains the traceback
+
```
Comment (jrbourbeau) [inline distributed/client.py:425]: Similar comment here, this returns the actual traceback object (or coroutine)

File: distributed/client.py [hunk 3]
```diff
@@ -362,11 +431,22 @@ def traceback(self, timeout=None, **kwargs):
...
 
     def release(self, _in_destructor=False):
-        # NOTE: this method can be called from different threads
-        # (see e.g. Client.get() or Future.__del__())
+        """
+
+        Parameters
+        ----------
+        _in_destructor: bool
+            Not used
```
Comment (jrbourbeau) [inline distributed/client.py:443]: If this isn't ever used, then I'd prefer to remove it altogether

File: distributed/client.py [hunk 4]
```diff
@@ -556,11 +685,20 @@ class Client(SyncMethodMixin):
     name: string (optional)
         Gives the client a name that will be included in logs generated on
         the scheduler for matters relating to this client
+    heartbeat_interval: int (optional)
+        Time in milliseconds between heartbeats to scheduler
+    serializers
+        The serializers to turn an object into a string
+    deserializers
+        The deserializers to turn the string into the original object
```
Comment (jrbourbeau) [inline distributed/client.py:693]: (De)Serializers don't convert to strings

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell @tharris72 -- apologies for the late review

**Pull request merged**